operator/controllers/watcher_controller.go

```diff
@@ -71,26 +74,35 @@ func (r *WatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if !watcherObj.DeletionTimestamp.IsZero() {
 		err := r.RemoveVirtualServiceConfigForCR(ctx, req.NamespacedName)
 		if err != nil {
-			return ctrl.Result{}, err
+			return ctrl.Result{RequeueAfter: r.RequeueIntervals.Failure}, err
 		}
 		controllerutil.RemoveFinalizer(watcherObj, watcherFinalizer)
-		return ctrl.Result{}, r.Update(ctx, watcherObj)
+		err = r.Update(ctx, watcherObj)
+		if err != nil {
+			return ctrl.Result{RequeueAfter: r.RequeueIntervals.Failure}, err
+		}
+		return ctrl.Result{}, nil
 	}
 
 	// check finalizer on native object
 	if !controllerutil.ContainsFinalizer(watcherObj, watcherFinalizer) {
 		controllerutil.AddFinalizer(watcherObj, watcherFinalizer)
-		return ctrl.Result{}, r.Update(ctx, watcherObj)
+		err := r.Update(ctx, watcherObj)
+		if err != nil {
+			return ctrl.Result{RequeueAfter: r.RequeueIntervals.Failure}, err
+		}
+		return ctrl.Result{}, nil
 	}
-
-	return ctrl.Result{RequeueAfter: r.RequeueIntervals.Failure},
-		r.UpdateVirtualServiceConfig(ctx, watcherObj)
+	err := r.UpdateVirtualServiceConfig(ctx, watcherObj)
+	if err != nil {
+		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Failure}, err
+	}
+	return ctrl.Result{RequeueAfter: r.RequeueIntervals.Success}, nil
 }
 
 func (r *WatcherReconciler) SetIstioClient() error {
```

Comment:
Same as for the Other setter I think you can simply make this entire call in SetupWithManager